clients/cobol-lsp-vscode-extension/src/services/LanguageClientService.ts

```diff
@@ -71,31 +67,26 @@ export class LanguageClientService {
 
     public async retrieveAnalysis(uri: string, text: string): Promise<any> {
         const languageClient = this.getLanguageClient();
-        await languageClient.onReady();
         return languageClient.sendRequest("extended/analysis", { uri, text });
     }
 
     public async invalidateConfiguration() {
         const languageClient = this.getLanguageClient();
-        await languageClient.onReady();
         languageClient.sendNotification(DidChangeConfigurationNotification.type, { settings: null });
     }
 
-    public start(): vscode.Disposable {
-        this.initHandlers();
-        return this.getLanguageClient().start();
+    public async start() {
+        const languageClient = this.getLanguageClient();
+        await languageClient.start().then(() => {
```

Comment:
I think it's better use await instead of then if we use async anyway

Reply:
updated